diff --git PKGBUILD PKGBUILD
index ffa9833..13dc071 100644
--- PKGBUILD
+++ PKGBUILD
@@ -12,11 +12,14 @@ license=(GPL-2.0-or-later)
 depends=(gd libarchive lua minizip qt6-base sdl2)
 makedepends=(cmake git glu mesa-libgl setconf scons)
 optdepends=('ffmpeg: for recording')
-source=("git+$url#tag=v$pkgver")
-b2sums=('df1b8706f3639c52ec5905a542052ee0eb7c5a727c9ee27af7c9f53b2b9f97b0445c6c6ff9f9504cef51a90cddc97b08b7f92ee7882de1bfb2baf879eac613c1')
+source=("git+$url#tag=v$pkgver"
+        'timestamp.patch')
+b2sums=('df1b8706f3639c52ec5905a542052ee0eb7c5a727c9ee27af7c9f53b2b9f97b0445c6c6ff9f9504cef51a90cddc97b08b7f92ee7882de1bfb2baf879eac613c1'
+        'f711fc790923cf12b57ecd6d40b2229b0cad787cf6a22b63119d90bae6d4c80ce3dcf920b5385a20485c7439ac1e1bdb745392d9a222fc38b16987f5a15bad93')
 
 prepare() {
   cd $pkgname
+  patch -p0 < ../timestamp.patch
   sed -i 's/-interim git//g' src/version.h
   setconf scripts/genGitHdr.sh GIT_URL "'""${source:4:34}""'"
   setconf scripts/genGitHdr.sh GIT_REV "${source#*=}"
@@ -30,7 +33,7 @@ build() {
     -D CMAKE_INSTALL_PREFIX=/usr \
     -D QT6=ON \
     -S $pkgname
-  make -C build
+  make -C build 
 }
 
 package() {
