--- src/utils/timeStamp.cpp
+++ src/utils/timeStamp.cpp
@@ -17,15 +17,19 @@
 #if defined(WIN32)
 #include <intrin.h>
 #pragma intrinsic(__rdtsc)
-#else
+#elif defined(__i386__) || defined(__x86_64__)
 #include <x86intrin.h>
 #endif
-
 static uint64_t rdtsc()
 {
-	return __rdtsc();
+#if defined(__riscv)
+    uint64_t val;
+    asm volatile("rdtime %0" : "=r"(val));
+    return val;
+#else
+    return __rdtsc();
+#endif
 }
-
 namespace FCEU
 {
    
